opencog/atomspace/BackingStore.cc

```diff
@@ -46,3 +48,13 @@ bool BackingStore::ignoreAtom(Handle h) const
 	return false;
 }
 
+void BackingStore::registerWith(AtomSpace* atomspace)
+{
+	atomspace->registerBackingStore(this);
```

Comment:
The alternative was to add the subclasses of AtomStorage as friends to all the places where AtomStorage was added, Atom and TLB etc.

Subclasses of friends can't see the same thing that the friend class can see, due to the way that C++ defines friends. So by adding protected accessors to AtomStorage the subclasses can now do what needs to be done without having to be explicitly added as friends.

This isn't so important with a couple of classes, but I suspect there will be future MySQL and SQLite classes as well.